allow pay period day to be selectable

bloGoms opened the pull request.

martin-the-geek commented:
None of the tests have been updated for the new API. Can you update the tests and add the appropriate new tests?

bloGoms committed:
allow pay period day to be selectable

allow pay period day to be selectable

allow pay period day to be selectable

allow pay period day to be selectable

martin-the-geek commented on src/pay-period-calendar/pay-period-calendar.component.spec.ts:
```diff
@@ -433,6 +433,58 @@ describe('PayPeriodCalendarComponent', () => {
         });
     });
 
+    describe('isSelected', () => {
```
this should describe 'isDaySelected'

martin-the-geek commented on src/pay-period-calendar/pay-period-selection.model.ts:
```diff
@@ -0,0 +1,6 @@
+
+export interface PayPeriodSelection {
+    payPeriod: PayPeriod;
+    dayIndex: number;
```
at some point we'll enable strict null checks, so let's make `dayIndex` be `number | null`.
